src/devices/README.md

```diff
@@ -251,6 +255,138 @@ iot/
           Foo.Tests.cs
 ```
 
+When creating a new binding, please follow the recommendations [from corefx coding guideliness](https://github.com/dotnet/corefx/tree/master/Documentation#coding-guidelines). On top, we recommend to use very descriptive namings, self explanatory names. Please make sure you don't use abbreviation. Describe any unit a sensor may use as well as the usual range of values. Check as well [the devices convention](../../Documentation/Devices-conventions.md). You will find below couple of examples of do and don't.
+
+### Using units
+
+| Do | Don't |
+|---|---|
+| Always use [UnitsNet](https://github.com/angularsen/UnitsNet) when possible: | Don't implement you own: |
+| ```public void SetEnvironmentData(Temperature temperature, Ratio humidity)``` | ```public void SetEnvironmentData(float temperature, float humidity)``` |
+| Temperature is a classic one, you have lots of available units available, [check here](https://github.com/angularsen/UnitsNet/tree/master/UnitsNet/GeneratedCode/Units) if anything exists before starting|In this case, there are existing units, use the existing one |
+
+| Do | Don't |
+|---|---|
+| Decribe your unit if it does not exist in [UnitsNet](https://github.com/angularsen/UnitsNet)|Leave an unkonwn measurement element |
+| ```public bool TryGetMyUnit(out double myUnitInUnitDescription)``` | ```public bool TryGetMyUnit(out double myUnit)``` |
+| In this case, you need to speciy in the variable name the unit used. Be as precise as possible | Never leave any unit without description in the naming if it's not an support UnitsNet unit. |
+
+| Do | Don't |
+|---|---|
+| Always add range of the units in the XML description|Never leave a unit without a full range description|
+| ```/// <param name="equivalentCO2">The equivalent CO2 (eCO2) output range for CCS811 is from 400ppm up to 29206ppm.</param>``` | ```/// <param name="equivalentCO2">The equivalent CO2 (eCO2)</param>``` |
+| Even when using the UnitsNet, make sure you give the usual range of a sensor value. | This makes it difficult to understand  which range the sensor is using. |
+
+### Naming of functions
+
+| Do |
+|---|
+| When reading a sensor and outputting data, use ```Try``` for example: |
+| ```public bool TryGetTemperature(out Temperature temperature)``` |
+| This makes it easy to understand that if you read and the retruned value is ```false``` the reading is not correct. And this means as well this function should not raise an exception when trying to read the sensor data. |
```

Comment:
You might want to add that exception in such cases are still ok in some cases, i.e.: IO errors or user error (basically if you have to try .. catch the whole method then it's an anti-pattern and likely you should just throw)